tst/suppressions/TSAN.supp

```diff
@@ -358,3 +358,100 @@ race:PeerConnectionFunctionalityTest_freeTurnDueToP2PFoundAfterTurnEstablished
+#     #16 executeDescribeSignalingState /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/StateMachine.c:346:5 (libkvsWebrtcSignalingClient.so+0x216f9)
+#     #17 stepStateMachine /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/amazon-kinesis-video-streams-pic/src/state/src/State.c:182:9 (libkvsWebrtcClient.so+0x757e5)
+#     #18 stepSignalingStateMachine /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/StateMachine.c:70:5 (libkvsWebrtcSignalingClient.so+0x23ad9)
+#     #19 executeGetTokenSignalingState /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/StateMachine.c:272:5 (libkvsWebrtcSignalingClient.so+0x2135a)
+#     #20 stepStateMachine /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/amazon-kinesis-video-streams-pic/src/state/src/State.c:182:9 (libkvsWebrtcClient.so+0x757e5)
+#     #21 stepSignalingStateMachine /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/StateMachine.c:70:5 (libkvsWebrtcSignalingClient.so+0x23ad9)
+#     #22 createSignalingSync /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/Signaling.c:151:5 (libkvsWebrtcSignalingClient.so+0x1c045)
+#     #23 com::amazonaws::kinesis::video::webrtcclient::SignalingApiFunctionalityTest_asyncIceConfigRefreshBeforeConnect_Test::TestBody() /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/tst/SignalingApiFunctionalityTest.cpp:2253:5 (webrtc_client_test+0x651aca)
+#     #24 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) <null> (webrtc_client_test+0x6ec65d)
+#
+#     Hint: use TSAN_OPTIONS=second_deadlock_stack=1 to get more informative warning message
+#
+#   Mutex M877774268853663472 acquired here while holding mutex M880307543644078960 in thread T572:
+#     #0 pthread_mutex_lock /tmp/final/llvm.src/projects/compiler-rt/lib/tsan/../sanitizer_common/sanitizer_common_interceptors.inc:4071:3 (webrtc_client_test+0x4fcb0e)
+#     #1 defaultLockMutex /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/amazon-kinesis-video-streams-pic/src/utils/src/Mutex.c:202:5 (libkvsWebrtcClient.so+0x6b9b8)
+#     #2 stepSignalingStateMachine /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/StateMachine.c:40:5 (libkvsWebrtcSignalingClient.so+0x23815)
+#     #3 refreshIceConfigurationCallback /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/src/source/Signaling/Signaling.c:593:9 (libkvsWebrtcSignalingClient.so+0x1ee93)
+#     #4 timerQueueExecutor /home/travis/build/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/open-source/amazon-kinesis-video-streams-pic/src/utils/src/TimerQueue.c:561:37 (libkvsWebrtcClient.so+0x72745)
```

Comment:
if timer dont hold execute lock while calling the callback we can avoid this deadlock. The executor thread can keep a local copy of timer tasks and update when necessary. Similar pattern in connectionListener executor

Reply:
I think it's doable. Release/re-acquire lock for the call. We should implement it soon.